lib/widgets/realm_content_network_image.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:flutter/material.dart';
+
+import '../api/core.dart';
+import 'store.dart';
+
+/// Like [Image.network], but includes [authHeader] iff `src` is on-realm.
+///
+/// Use this to present image content in the ambient realm: avatars, images in
+/// messages, etc. Must have a [PerAccountStoreWidget] ancestor.
+///
+/// If `src` is an on-realm URL (it has the same origin as the ambient
+/// [Auth.realmUrl]), then an HTTP request to fetch the image will include the
+/// user's [authHeader].
+///
+/// If `src` is off-realm (e.g., a Gravatar URL), no auth header will be sent.
+///
+/// Does not limit cache access by realm; caching behavior is defined by
+/// [Image.network], which this uses.
```

Comment:
`Image.network` will cache network images, to avoid making unnecessary network requests. Here's one note from its [doc](https://api.flutter.dev/flutter/widgets/Image/Image.network.html) about the caching behavior:

> All network images are cached regardless of HTTP headers.

My thought was to avoid implying (mainly in its name, I guess) that our widget has some extra realm-aware caching logic on top of that. One way such logic could look is to have each realm use a separate cache, so that the presence, absence, or details of a rendered image don't depend on what other realm you've seen the image's URL in.

With the current implementation, I could imagine some user-visible surprises. Take an authenticated image URL on Realm A:

- First load it on Realm A; the image appears there. Then try to view it on Realm B…and it still appears, from the cache, even though your Realm B credentials would've been rejected.
- First load it on Realm B; the image doesn't appear there because your credentials are rejected. Then try to view it on Realm A…if the cached response is used directly without a re-request with your Realm A credentials, then the image still won't appear, even though you should have access.

Reply:
I see.  Maybe this, then:
```dart
/// The image will be cached according to the cache behavior of [Image.network],
/// which may mean the cache is shared between realms.
```

FWIW I think the "regardless of HTTP headers" there is not meant to say that headers don't change the cache key (though they may not), but rather that it ignores the HTTP headers that are supposed to control caching.

Reply:
Makes sense; I'll do that.